daffodil-japi/src/main/scala/org/apache/daffodil/japi/Daffodil.scala

```diff
@@ -490,7 +490,11 @@ class DataLocation private[japi] (dl: SDataLocation) {
   override def toString() = dl.toString
 
   /**
-   * Determine if this data location is at the end of the input data
+   * Determine if this data location is known to be at the end of the data stream.
+   *
+   * Note that this can be false, yet all bits consumed. This occurs if no
+   * subsequent input read operation from the data occurred causing the
+   * input stream to detect end-of-data/EOF.
```

Comment:
This is an API change. I can imagine cases where people relied on this behavior for checking left over data, and I feel like it's very easy for binary formats to never have this set to true. Prevously, unless you were using a streaming thing, we always knew for sure if we hit EOD.

Maybe we need a tunable to switch between the "read one byte to figured out EOD" vs "must have hit EOD before, and we aren't actually sure if we are at EOD yet"? We can default to the old behavior, but people streaming or doing TCP stuff can switch to the new behavior, with the understanding that isAtEnd might not be accurate?

Reply:
I suggest we deprecate isAtEnd, but give it the bad-old functionality.